File: baekjoon/p06996/Main.java
package baekjoon.p06996;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.next(), sc.next()));
        }
    }

    private static String solve(String a, String b) {
        int size = 26;
        int[] cntA = new int[size];
        int[] cntB = new int[size];

        for (char ch : a.toCharArray()) {
            int idx = ch - 'a';
            cntA[idx]++;
        }

        for (char ch : b.toCharArray()) {
            int idx = ch - 'a';
            cntB[idx]++;
        }

        for (int i = 0; i < size; i++) {
            if (cntA[i] != cntB[i]) {
                return String.format("%s & %s are NOT anagrams.", a, b);
            }
        }

        return String.format("%s & %s are anagrams.", a, b);
    }

}

File: baekjoon/p06996/Main.kt
package baekjoon.p06996

fun main() {
    repeat(readLine()!!.toInt()) {
        val inputs = readLine()!!.split(" ")
        println(solve(inputs[0], inputs[1]))
    }
}

private fun solve(a: String, b: String): String {
    val size = 26
    val cntA = IntArray(size)
    val cntB = IntArray(size)

    for (i in 0 until a.length) {
        val idx = a[i] - 'a'
        cntA[idx]++
    }

    for (i in 0 until b.length) {
        val idx = b[i] - 'a'
        cntB[idx]++
    }

    for (i in 0 until size) {
        if (cntA[i] != cntB[i]) {
            return "$a & $b are NOT anagrams."
        }
    }

    return "$a & $b are anagrams."
}